client/ayon_sitesync/launch_hooks/pre_copy_last_published_workfile.py

```diff
@@ -102,40 +95,47 @@ def execute(self):
             )
             return
 
-        max_retries = int((sync_server.sync_project_settings[project_name]
-                                                            ["config"]
-                                                            ["retry_cnt"]))
+        max_retries = int((sitesync_addon.sync_project_settings[project_name]
+                                                               ["config"]
+                                                               ["retry_cnt"]))
 
         self.log.info("Trying to fetch last published workfile...")
 
-        asset_doc = self.data.get("asset_doc")
+        folder_entity = self.data.get("folder_entity")
         anatomy = self.data.get("anatomy")
-
+        hierarchy = "/".join(folder_entity["path"].split("/")[:-1]).lstrip("/")
         context_filters = {
-            "asset": asset_name,
-            "family": "workfile",
-            "task": {"name": task_name, "type": task_type}
+            "folder.name": [folder_entity["name"]],
+            "hierarchy": [hierarchy],
+            "product.type": ["workfile"],
+            "task.name": [task_name],
+            "task.type": [task_type]
         }
 
-        workfile_representations = list(get_representations(
+        workfile_representation_ids = list(get_repre_ids_by_context_filters(
             project_name,
             context_filters=context_filters
         ))
 
-        if not workfile_representations:
+        if not workfile_representation_ids:
             self.log.debug(
                 'No published workfile for task "{}" and host "{}".'.format(
                     task_name, host_name
                 )
             )
             return
 
-        filtered_repres = filter(
-            lambda r: r["context"].get("version") is not None,
-            workfile_representations
-        )
+        workfile_representations = get_representations(
+            project_name, workfile_representation_ids)
+
+        if not workfile_representations:
+            self.log.debug(
+                f"Not found representations for '{workfile_representation_ids}"
+            )
+            return
+
         workfile_representation = max(
-            filtered_repres, key=lambda r: r["context"]["version"]
+            workfile_representations, key=lambda r: r["context"]["version"]
```

Comment:
This will crash on hero versions. Probably should be fetched version entities by `"versionId"` on representations.

Reply:
There is a hero version on workfile?

Reply:
There can be hero version... Also using data from representation context should not happen, in ideal case.